src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -317,7 +323,24 @@ bool GCToOSInterface::Initialize()
 
     NUMASupportInitialize();
 
-    return true;
+#ifdef TARGET_APPLE
+    const char* mem_free_name = "kern.memorystatus_level";
+    int rc = sysctlnametomib(mem_free_name, NULL, &g_kern_memorystatus_level_mib_length);
+    if (rc == 0)
+    {
+        g_kern_memorystatus_level_mib = (int*)malloc(g_kern_memorystatus_level_mib_length * sizeof(int));
```

Comment:
presumably if this returns null then `sysctlnametomib` would return a failure?

Reply:
Thank you for spotting the missing check, fixed.